exchanges/bitfinex/bitfinex_websocket.go

```diff
@@ -469,9 +423,80 @@ func (b *Bitfinex) wsHandleData(respRaw []byte) error {
+	event, err := jsonparser.GetUnsafeString(respRaw, "event")
+	if err != nil {
+		return fmt.Errorf("error parsing WS event name: %w from message: %s", err, respRaw)
+	}
+	switch event {
+	case wsEventSubscribed:
+		subID, err := jsonparser.GetUnsafeString(respRaw, "subId")
+		if err != nil {
+			return fmt.Errorf("error parsing WS subscribed event subId: %w from message: %s", err, respRaw)
+		}
+		if !b.Websocket.Match.IncomingWithData("subscribe:"+subID, respRaw) {
+			return fmt.Errorf("%v channel subscribe listener not found", subID)
+		}
+	case wsEventUnsubscribed:
+		chanID, err := jsonparser.GetUnsafeString(respRaw, "chanId")
+		if err != nil {
+			return fmt.Errorf("error parsing WS unsubscribed event chanId: %w from message: %s", err, respRaw)
+		}
+		if !b.Websocket.Match.IncomingWithData("unsubscribe:"+chanID, respRaw) {
+			return fmt.Errorf("%v channel unsubscribe listener not found", chanID)
+		}
+	case wsEventError:
+		if subID, err := jsonparser.GetUnsafeString(respRaw, "subId"); err == nil {
+			if !b.Websocket.Match.IncomingWithData("subscribe:"+subID, respRaw) {
+				return fmt.Errorf("%v channel subscribe listener not found", subID)
+			}
+		} else if chanID, err := jsonparser.GetUnsafeString(respRaw, "chanId"); err == nil {
+			if !b.Websocket.Match.IncomingWithData("unsubscribe:"+chanID, respRaw) {
+				return fmt.Errorf("%v channel unsubscribe listener not found", chanID)
+			}
+		} else {
+			return fmt.Errorf("unknown channel error; Message: %s", respRaw)
+		}
+	case wsEventAuth:
+		status, err := jsonparser.GetUnsafeString(respRaw, "status")
+		if err != nil {
+			return fmt.Errorf("error parsing WS auth event status: %w from message: %s", err, respRaw)
+		}
+		if status == "OK" {
+			var glob map[string]interface{}
+			if err := json.Unmarshal(respRaw, &glob); err != nil {
+				return fmt.Errorf("unable to Unmarshal auth resp; Error: %w Msg: %v", err, respRaw)
+			}
+			// TODO - Send a better value down the channel
+			b.Websocket.DataHandler <- glob
+		} else {
+			errCode, err := jsonparser.GetInt(respRaw, "code")
+			if err != nil {
+				log.Errorf(log.ExchangeSys, "%s error parsing WS auth event error code: %s", b.Name, err)
+			}
+			return fmt.Errorf("WS auth subscription error; Status: %s Error Code: %d", status, errCode)
+		}
+	case wsEventInfo:
+		// Nothing to do with info for now.
+		// version or platform.status might be useful in the future.
+	case wsEventConf:
+		status, err := jsonparser.GetUnsafeString(respRaw, "status")
+		if err != nil {
+			return fmt.Errorf("error parsing WS configure channel event status: %w from message: %s", err, respRaw)
+		}
+		if status != "OK" {
+			return fmt.Errorf("WS configure channel error; Status: %s", status)
+		}
+	default:
+		return fmt.Errorf("unknown WS event msg: %s", respRaw)
+	}
+
+	return nil
+}
+
+func (b *Bitfinex) handleWSChannelUpdate(c *stream.ChannelSubscription, eventType string, d []interface{}) error {
```

Comment:
No problem. I'll replicate and fix today.

Reply:
Can't replicate:
`error - updating orderbook error: orderbook depth not found for Exchange Bitfinex CurrencyPair: BTC-USD AssetType: spot`

but I am seeing the same second batch of subscriptions happening which is causing dup errors:
`subscription failure: subscribe: dup (code: 10301); Channel: book Pair: BTC-USD`
I'm going to chase this down now.

Can you still replicate the orderbook depth issue?

Reply:
Huh. So ... Duplicates is because `WebsocketRoutineManager.websocketRoutine` calls FlushChannels.
Comparing to master, that no longer happens as a result of Ryan's [recent change](https://github.com/thrasher-corp/gocryptotrader/commit/773441d5a7cd1323b1f2b28be1ad77233962e663)

So I'll rebase this and re-test.

Reply:
Yeah I haven't received the depth not found error this morning 🤙 Just the dupe likely a result of margin and spot having the same pair enabled

Reply:
Confirming all looks good after rebase on master